packages/pg/lib/client.js

```diff
@@ -545,6 +545,15 @@ Client.prototype.query = function (config, values, callback) {
 Client.prototype.end = function (cb) {
   this._ending = true
 
+  // if we have never connected, then end is a noop, callback immediately
+  if (this.connection.stream.readyState === 'closed') {
+    if (cb) {
+      cb()
+    } else {
+      return Promise.resolve()
```

Comment:
For consistency while this feature exists:

```suggestion
      return this._Promise.resolve()
```

Reply:
yeah good call!